broker.go

```diff
@@ -38,23 +20,41 @@ const (
 	DefaultFailureSleep = 100 * time.Millisecond
 )
 
+// Broker represents an InfluxDB specific messaging broker.
+type Broker struct {
+	*messaging.Broker
+
+	done chan struct{}
+
+	// send CQ processing requests to the same data node
+	// currentCQProcessingNode *messaging.Replica // FIX(benbjohnson)
+
+	// variables to control when to trigger processing and when to timeout
+	TriggerInterval     time.Duration
+	TriggerTimeout      time.Duration
+	TriggerFailurePause time.Duration
+}
+
 // NewBroker returns a new instance of a Broker with default values.
 func NewBroker() *Broker {
-	b := &Broker{
+	return &Broker{
+		Broker:              messaging.NewBroker(),
 		TriggerInterval:     5 * time.Second,
 		TriggerTimeout:      20 * time.Second,
 		TriggerFailurePause: 1 * time.Second,
 	}
-	b.Broker = messaging.NewBroker()
-	return b
 }
 
 // RunContinuousQueryLoop starts running continuous queries on a background goroutine.
 func (b *Broker) RunContinuousQueryLoop() {
-	b.done = make(chan struct{})
-	go b.continuousQueryLoop(b.done)
+	// FIX(benbjohnson)
+	// b.done = make(chan struct{})
```

Comment:
I remember Ben saying something about the brokers not knowing about the data nodes in this new branch, but I think that they should. The data nodes have to connect to get replication so they should include a connect string so that the raft leader can send requests down to the data nodes.

The brokers will need to know about the data nodes anyway so that they can redirect a data node that requests a truncated topic to another data node that can give them a copy of the entire thing.